vcpkg/src/lib.rs

```diff
@@ -398,19 +398,196 @@ fn find_vcpkg_target(cfg: &Config, target_triplet: &TargetTriplet) -> Result<Vcp
     let lib_path = base.join("lib");
     let bin_path = base.join("bin");
     let include_path = base.join("include");
+    let packages_path = vcpkg_root.join("packages");
 
     Ok(VcpkgTarget {
-        vcpkg_triplet: target_triplet.triplet.to_owned(),
         lib_path: lib_path,
         bin_path: bin_path,
         include_path: include_path,
-        is_static: target_triplet.is_static,
         status_path: status_path,
-        lib_suffix: target_triplet.lib_suffix.to_owned(),
-        strip_lib_prefix: target_triplet.strip_lib_prefix,
+        packages_path: packages_path,
+        target_triplet: target_triplet.clone(),
     })
 }
 
+/// Parsed knowledge from a .pc file.
+#[derive(Debug)]
+struct PcFile {
+    /// The pkg-config name of this library.
+    id: String,
+    /// List of libraries found as '-l', translated to a given vcpkg_target. e.g. libbrotlicommon.a
+    libs: Vec<String>,
+    /// List of pkgconfig dependencies, e.g. PcFile::id.
+    deps: Vec<String>,
+}
+impl PcFile {
+    fn parse_pc_file(vcpkg_target: &VcpkgTarget, path: &Path) -> Result<PcFile, Error> {
+        // Extract the pkg-config name.
+        let id = try!(path
+            .file_stem()
+            .ok_or_else(|| Error::VcpkgInstallation(format!(
+                "pkg-config file {} has bogus name",
+                path.to_string_lossy()
+            ))))
+        .to_string_lossy();
+        // Read through the file and gather what we want.
+        let mut file = try!(File::open(path)
+            .map_err(|_| Error::VcpkgInstallation(format!("Couldn't open {}", path.display()))));
+        let mut pc_file_contents = String::new();
+
+        try!(file
+            .read_to_string(&mut pc_file_contents)
+            .map_err(|_| Error::VcpkgInstallation(format!("Couldn't read {}", path.display()))));
+        PcFile::from_str(&id, &pc_file_contents, &vcpkg_target.target_triplet)
+    }
+    fn from_str(id: &str, s: &str, target_triplet: &TargetTriplet) -> Result<Self, Error> {
+        let mut libs = Vec::new();
+        let mut deps = Vec::new();
+
+        for line in s.lines() {
+            // We could collect alot of stuff here, but we only care about Requires and Libs for the moment.
+            if line.starts_with("Requires:") {
+                let requires_args = line.split(":").skip(1).next().unwrap_or("").split(",");
+                for mut dep in requires_args {
+                    // Drop any versioning requirements, we only care about library order and rely upon
+                    // port dependencies to resolve versioning.
+                    if let Some(inequality) = dep.find(|c| c == '=' || c == '<' || c == '>') {
+                        dep = dep.split_at(inequality).0;
+                    }
+                    dep = dep.trim();
+                    // Sometimes Requires: lines don't have commas, so split again.
+                    deps.extend(dep.split_whitespace().map(|s| s.to_owned()));
```

Comment:
I think the parsing here will have some problems when there are missing commas. Consider:

```
Requires: foo >= 1 bar baz
```

As-is, this algorithm would lose the `bar baz` dependencies altogether. I think this needs to split the string into pieces using either commas or whitespace as a separator, and then walk through the resulting list adding items as it goes. If a token is a relational operator, it and the next token should be skipped.

Here are some example Requires lines from my system — might be good to set up testing to make sure they're all handled correctly:

```
Requires: cairo libpng
Requires: cairo xcb >= 1.6 xcb-render >= 1.6
Requires: glib-2.0, gobject-2.0
Requires: glib-2.0 >=  2.58.0, gobject-2.0 >=  2.58.0
```

Reply:
Ya what I shared isn't without problems.

I rewrote it using your examples as unittests with your suggestions. Looks a lot better.